cooltools/sandbox/cooler_filters/pixel_filter_util.py

```diff
@@ -0,0 +1,176 @@
+            if store_name in grp["bins"]:
+                del grp["bins"][store_name]
+            h5opts = dict(compression="gzip", compression_opts=6)
+            grp["bins"].create_dataset(store_name, data=bin_mask, **h5opts, dtype=bool)
+
+    return bin_mask
+
+
+def _pixel_filter(chunk_pixels, good_bins_index):
+    """
+    Filters a chunk of pixels based on a list of good bin indices.
+
+    Parameters
+    ----------
+    chunk_pixels : pandas.DataFrame
+        A DataFrame containing the pixels to be filtered. It must have columns 'bin1_id' and 'bin2_id'.
+    good_bins_index : list of int
+        A list of indices representing the good bins.
+
+    Returns
+    -------
+    pandas.DataFrame
+        A DataFrame containing only the pixels whose bin1_id and bin2_id are in good_bins_index.
+    """
+
+    pixels_mask = chunk_pixels["bin1_id"].isin(good_bins_index) * chunk_pixels[
+        "bin2_id"
+    ].isin(good_bins_index)
+    return chunk_pixels[pixels_mask]
+
+
+def pixel_iter_chunks(clr, chunksize):
+    """
+    Iterate over the pixels of a cooler object in chunks of a given size.
+
+    Parameters
+    ----------
+    clr : cooler.Cooler
+        A cooler object containing Hi-C data.
+    chunksize : int
+        The size of each chunk of pixels to iterate over.
+
+    Yields
+    ------
+    chunk : numpy.ndarray
+        A chunk of pixels of size `chunksize`.
+    """
+    selector = clr.pixels()
+    for lo, hi in cooler.util.partition(0, len(selector), chunksize):
+        chunk = selector[lo:hi]
+        yield chunk
+
+@pool_decorator
+def create_filtered_cooler(
+    output_uri, clr, bin_mask, chunksize=10_000_000, nproc=1, map=map
+):
+    """
+    Create a filtered cooler file from a given cooler object and a binary mask of good bins.
+
+    Parameters
+    ----------
+    output_uri : str
+        The URI of the output cooler file to be created.
+    clr : cooler.Cooler
+        The cooler object to be filtered.
+    bin_mask : numpy.ndarray
+        A boolean array indicating which bins to keep (True) and which to discard (False).
+        Must have the same length as the number of bins in the cooler object.
+    nproc : int, optional
+        The number of processes to use for parallelization. Default is 16.
+    chunksize : int, optional
+        The number of pixels to process per chunk. Default is 10,000,000.
+
+    Returns
+    -------
+    None
+    """
+    if len(bin_mask) != clr.bins().shape[0]:
+        raise ValueError(
+            "bin_mask should have the same length as bin table in cool file"
+        )
+    logger.debug("Start to create cooler file...")
+    bin_table = clr.bins()[:]
+    good_bins_index = np.array(range(clr.bins().shape[0]))[bin_mask]
+    pixels_filter = functools.partial(_pixel_filter, good_bins_index=good_bins_index)
+
+    cooler.create_cooler(
+        output_uri,
+        bins=bin_table,
+        pixels=map(pixels_filter, pixel_iter_chunks(clr, chunksize)),
```

Comment:
You probably want either builtin `map`, which is lazy and returns an iterator, or `mp.Pool.imap` + `ordered=True`, or potentially `mp.Pool.imap_unordered` + `ordered=False` (this does a 2-pass creation algorithm).